Index: lcl/interfaces/wince/wincecallback.inc
===================================================================
--- lcl/interfaces/wince/wincecallback.inc	(revision 11105)
+++ lcl/interfaces/wince/wincecallback.inc	(working copy)
@@ -1127,6 +1127,24 @@
             CBN_CLOSEUP:
               UpdateComboBoxText(TCustomComboBox(lWinControl));
           end;
+        { WINCE menubar button }
+        if (WindowInfo^.WinControl is TCustomForm) then
+           case lo(WParam) of
+             { OK button }
+             IDOK: begin
+                if (TCustomForm(WindowInfo^.WinControl).DefaultControl<>nil) then
+                   TCustomForm(WindowInfo^.WinControl).DefaultControl.ExecuteDefaultAction
+                else  
+                   TCustomForm(WindowInfo^.WinControl).ModalResult:=mrOK;
+             end;
+             { Cancel button }
+             IDCANCEL: begin
+                if (TCustomForm(WindowInfo^.WinControl).CancelControl<>nil) then
+                   TCustomForm(WindowInfo^.WinControl).CancelControl.ExecuteCancelAction
+                else   
+                   TCustomForm(WindowInfo^.WinControl).ModalResult:=mrCancel;
+             end;
+           end;
       end;
 
       // no specific message found? try send a general msg
