Index: Packages/GLScene_DesignTime.lpk
===================================================================
--- Packages/GLScene_DesignTime.lpk	(revision 7332)
+++ Packages/GLScene_DesignTime.lpk	(working copy)
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <CONFIG>
-  <Package Version="4">
+  <Package Version="5">
     <Name Value="GLScene_DesignTime"/>
     <Type Value="DesignTime"/>
     <AddToProjectUsesSection Value="True"/>
@@ -8,7 +8,7 @@
       <Version Value="11"/>
       <SearchPaths>
         <IncludeFiles Value="../Source"/>
-        <OtherUnitFiles Value="../Source"/>
+        <OtherUnitFiles Value="../Source;../Source/Basis;../Source/DesignTime;../Source/Plateform;../Source/GameAPIs;../Source/Shaders;../Source/SoundVideoAPIs;../Source/FileFormats"/>
         <UnitOutputDirectory Value="lib/$(TargetCPU)-$(TargetOS)"/>
         <SrcPath Value="../Lazarus"/>
       </SearchPaths>
@@ -27,7 +27,7 @@
     <Version Major="1"/>
     <Files Count="21">
       <Item1>
-        <Filename Value="../Source/FXCollectionEditor.pas"/>
+        <Filename Value="../Source/DesignTime/FXCollectionEditor.pas"/>
         <UnitName Value="FXCollectionEditor"/>
       </Item1>
       <Item2>
@@ -39,80 +39,81 @@
         <Type Value="LRS"/>
       </Item3>
       <Item4>
-        <Filename Value="../Source/GLSceneRegister.pas"/>
+        <Filename Value="../Source/DesignTime/GLSceneRegister.pas"/>
         <HasRegisterProc Value="True"/>
         <UnitName Value="GLSceneRegister"/>
       </Item4>
       <Item5>
-        <Filename Value="../Source/GLObjectManager.pas"/>
+        <Filename Value="../Source/Basis/GLObjectManager.pas"/>
         <UnitName Value="GLObjectManager"/>
       </Item5>
       <Item6>
-        <Filename Value="../Source/FLibMaterialPicker.pas"/>
+        <Filename Value="../Source/DesignTime/FLibMaterialPicker.pas"/>
         <UnitName Value="FLibMaterialPicker"/>
       </Item6>
       <Item7>
-        <Filename Value="../Source/FMaterialEditorForm.pas"/>
+        <Filename Value="../Source/DesignTime/FMaterialEditorForm.pas"/>
         <UnitName Value="FMaterialEditorForm"/>
       </Item7>
       <Item8>
-        <Filename Value="../Source/FRColorEditor.pas"/>
+        <Filename Value="../Source/DesignTime/FRColorEditor.pas"/>
         <UnitName Value="FRColorEditor"/>
       </Item8>
       <Item9>
-        <Filename Value="../Source/FRFaceEditor.pas"/>
+        <Filename Value="../Source/DesignTime/FRFaceEditor.pas"/>
         <UnitName Value="FRFaceEditor"/>
       </Item9>
       <Item10>
-        <Filename Value="../Source/FRMaterialPreview.pas"/>
+        <Filename Value="../Source/DesignTime/FRMaterialPreview.pas"/>
         <UnitName Value="FRMaterialPreview"/>
       </Item10>
       <Item11>
-        <Filename Value="../Source/FRTextureEdit.pas"/>
+        <Filename Value="../Source/DesignTime/FRTextureEdit.pas"/>
         <UnitName Value="FRTextureEdit"/>
       </Item11>
       <Item12>
-        <Filename Value="../Source/FRTrackBarEdit.pas"/>
+        <Filename Value="../Source/DesignTime/FRTrackBarEdit.pas"/>
         <UnitName Value="FRTrackBarEdit"/>
       </Item12>
       <Item13>
-        <Filename Value="../Source/FVectorEditor.pas"/>
+        <Filename Value="../Source/DesignTime/FVectorEditor.pas"/>
         <UnitName Value="FVectorEditor"/>
       </Item13>
       <Item14>
-        <Filename Value="../Source/FInfo.pas"/>
+        <Filename Value="../Source/DesignTime/FInfo.pas"/>
         <UnitName Value="FInfo"/>
       </Item14>
       <Item15>
-        <Filename Value="../Source/FGUILayoutEditor.lfm"/>
+        <Filename Value="../Source/DesignTime/FGUILayoutEditor.lfm"/>
         <Type Value="LFM"/>
       </Item15>
       <Item16>
-        <Filename Value="../Source/FGUILayoutEditor.lrs"/>
+        <Filename Value="../Source/DesignTime/FGUILayoutEditor.lrs"/>
         <Type Value="LRS"/>
       </Item16>
       <Item17>
-        <Filename Value="../Source/FGUILayoutEditor.pas"/>
+        <Filename Value="../Source/DesignTime/FGUILayoutEditor.pas"/>
         <UnitName Value="FGUILayoutEditor"/>
       </Item17>
       <Item18>
-        <Filename Value="../Source/FGLSceneEdit.pas"/>
+        <Filename Value="../Source/DesignTime/FGLSceneEdit.pas"/>
         <UnitName Value="FGLSceneEdit"/>
       </Item18>
       <Item19>
-        <Filename Value="../Source/FShaderUniformEditor.pas"/>
+        <Filename Value="../Source/DesignTime/FShaderUniformEditor.pas"/>
         <UnitName Value="FShaderUniformEditor"/>
       </Item19>
       <Item20>
-        <Filename Value="../Source/FGUISkinEditor.pas"/>
+        <Filename Value="../Source/DesignTime/FGUISkinEditor.pas"/>
         <UnitName Value="FGuiSkinEditor"/>
       </Item20>
       <Item21>
-        <Filename Value="../Source/RegisterXCollection.pas"/>
+        <Filename Value="../Source/DesignTime/RegisterXCollection.pas"/>
         <AddToUsesPkgSection Value="False"/>
         <UnitName Value="RegisterXCollection"/>
       </Item21>
     </Files>
+    <CompatibilityMode Value="True"/>
     <i18n>
       <EnableI18N Value="True"/>
       <OutDir Value="../Locale/Languages/Lazarus"/>
@@ -142,7 +143,6 @@
     </UsageOptions>
     <PublishOptions>
       <Version Value="2"/>
-      <IgnoreBinaries Value="False"/>
     </PublishOptions>
   </Package>
 </CONFIG>
Index: Packages/GLScene_RunTime.lpk
===================================================================
--- Packages/GLScene_RunTime.lpk	(revision 7332)
+++ Packages/GLScene_RunTime.lpk	(working copy)
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <CONFIG>
-  <Package Version="4">
+  <Package Version="5">
     <Name Value="GLScene_RunTime"/>
     <AddToProjectUsesSection Value="True"/>
     <CompilerOptions>
@@ -7,7 +7,7 @@
       <Version Value="11"/>
       <SearchPaths>
         <IncludeFiles Value="../Source"/>
-        <OtherUnitFiles Value="../Source"/>
+        <OtherUnitFiles Value="../Source;../Source/Basis;../Source/FileFormats;../Source/Shaders;../Source/GameAPIs;../Source/ScriptingAPIs;../Source/Plateform;../Source/DesignTime"/>
         <UnitOutputDirectory Value="lib/$(TargetCPU)-$(TargetOS)"/>
       </SearchPaths>
       <Parsing>
@@ -33,17 +33,17 @@
     <Description Value="GLScene OpenGL 3D library
 "/>
     <Version Major="1"/>
-    <Files Count="238">
+    <Files Count="240">
       <Item1>
-        <Filename Value="../Source/GLApplicationFileIO.pas"/>
+        <Filename Value="../Source/Basis/GLApplicationFileIO.pas"/>
         <UnitName Value="GLApplicationFileIO"/>
       </Item1>
       <Item2>
-        <Filename Value="../Source/GLAsyncTimer.pas"/>
+        <Filename Value="../Source/Basis/GLAsyncTimer.pas"/>
         <UnitName Value="GLAsyncTimer"/>
       </Item2>
       <Item3>
-        <Filename Value="../Source/GLBaseClasses.pas"/>
+        <Filename Value="../Source/Basis/GLBaseClasses.pas"/>
         <UnitName Value="GLBaseClasses"/>
       </Item3>
       <Item4>
@@ -55,15 +55,15 @@
         <UnitName Value="GLCurvesAndSurfaces"/>
       </Item5>
       <Item6>
-        <Filename Value="../Source/GLGeometryBB.pas"/>
+        <Filename Value="../Source/Basis/GLGeometryBB.pas"/>
         <UnitName Value="GLGeometryBB"/>
       </Item6>
       <Item7>
-        <Filename Value="../Source/GLGeometryCoordinates.pas"/>
+        <Filename Value="../Source/Basis/GLGeometryCoordinates.pas"/>
         <UnitName Value="GLGeometryCoordinates"/>
       </Item7>
       <Item8>
-        <Filename Value="../Source/GLCanvas.pas"/>
+        <Filename Value="../Source/Basis/GLCanvas.pas"/>
         <UnitName Value="GLCanvas"/>
       </Item8>
       <Item9>
@@ -71,7 +71,7 @@
         <UnitName Value="GLColor"/>
       </Item9>
       <Item10>
-        <Filename Value="../Source/GLCoordinates.pas"/>
+        <Filename Value="../Source/Basis/GLCoordinates.pas"/>
         <UnitName Value="GLCoordinates"/>
       </Item10>
       <Item11>
@@ -111,15 +111,15 @@
         <UnitName Value="GLMeshUtils"/>
       </Item19>
       <Item20>
-        <Filename Value="../Source/GLOctree.pas"/>
+        <Filename Value="../Source/Basis/GLOctree.pas"/>
         <UnitName Value="GLOctree"/>
       </Item20>
       <Item21>
-        <Filename Value="../Source/OpenGLAdapter.pas"/>
+        <Filename Value="../Source/Basis/OpenGLAdapter.pas"/>
         <UnitName Value="OpenGLAdapter"/>
       </Item21>
       <Item22>
-        <Filename Value="../Source/OpenGLTokens.pas"/>
+        <Filename Value="../Source/Basis/OpenGLTokens.pas"/>
         <UnitName Value="OpenGLTokens"/>
       </Item22>
       <Item23>
@@ -131,7 +131,7 @@
         <UnitName Value="GLPersistentClasses"/>
       </Item24>
       <Item25>
-        <Filename Value="../Source/GLPolynomials.pas"/>
+        <Filename Value="../Source/Basis/GLPolynomials.pas"/>
         <UnitName Value="GLPolynomials"/>
       </Item25>
       <Item26>
@@ -139,187 +139,187 @@
         <UnitName Value="GLSpacePartition"/>
       </Item26>
       <Item27>
-        <Filename Value="../Source/GLSpline.pas"/>
+        <Filename Value="../Source/Basis/GLSpline.pas"/>
         <UnitName Value="GLSpline"/>
       </Item27>
       <Item28>
-        <Filename Value="../Source/GLVectorGeometry.pas"/>
+        <Filename Value="../Source/Basis/GLVectorGeometry.pas"/>
         <UnitName Value="GLVectorGeometry"/>
       </Item28>
       <Item29>
-        <Filename Value="../Source/GLVectorLists.pas"/>
+        <Filename Value="../Source/Basis/GLVectorLists.pas"/>
         <UnitName Value="GLVectorLists"/>
       </Item29>
       <Item30>
-        <Filename Value="../Source/GLVectorTypes.pas"/>
+        <Filename Value="../Source/Basis/GLVectorTypes.pas"/>
         <UnitName Value="GLVectorTypes"/>
       </Item30>
       <Item31>
-        <Filename Value="../Source/GLVerletTypes.pas"/>
+        <Filename Value="../Source/Basis/GLVerletTypes.pas"/>
         <UnitName Value="GLVerlettypes"/>
       </Item31>
       <Item32>
-        <Filename Value="../Source/GLVerletHairClasses.pas"/>
+        <Filename Value="../Source/Basis/GLVerletHairClasses.pas"/>
         <UnitName Value="GLVerletHairClasses"/>
       </Item32>
       <Item33>
-        <Filename Value="../Source/GLXCollection.pas"/>
+        <Filename Value="../Source/Basis/GLXCollection.pas"/>
         <UnitName Value="GLXCollection"/>
       </Item33>
       <Item34>
-        <Filename Value="../Source/XOpenGL.pas"/>
+        <Filename Value="../Source/Basis/XOpenGL.pas"/>
         <UnitName Value="XOpenGL"/>
       </Item34>
       <Item35>
-        <Filename Value="../Source/Const3DS.pas"/>
+        <Filename Value="../Source/FileFormats/Const3DS.pas"/>
         <UnitName Value="Const3DS"/>
       </Item35>
       <Item36>
-        <Filename Value="../Source/DDSImage.pas"/>
+        <Filename Value="../Source/FileFormats/DDSImage.pas"/>
         <UnitName Value="DDSImage"/>
       </Item36>
       <Item37>
-        <Filename Value="../Source/DXTC.pas"/>
+        <Filename Value="../Source/FileFormats/DXTC.pas"/>
         <UnitName Value="DXTC"/>
       </Item37>
       <Item38>
-        <Filename Value="../Source/File3DS.pas"/>
+        <Filename Value="../Source/FileFormats/File3DS.pas"/>
         <UnitName Value="File3DS"/>
       </Item38>
       <Item39>
-        <Filename Value="../Source/FileB3D.pas"/>
+        <Filename Value="../Source/FileFormats/FileB3D.pas"/>
         <UnitName Value="FileB3D"/>
       </Item39>
       <Item40>
-        <Filename Value="../Source/FileGL2.pas"/>
+        <Filename Value="../Source/FileFormats/FileGL2.pas"/>
         <UnitName Value="FileGL2"/>
       </Item40>
       <Item41>
-        <Filename Value="../Source/FileMD2.pas"/>
+        <Filename Value="../Source/FileFormats/FileMD2.pas"/>
         <UnitName Value="FileMD2"/>
       </Item41>
       <Item42>
-        <Filename Value="../Source/FileMD3.pas"/>
+        <Filename Value="../Source/FileFormats/FileMD3.pas"/>
         <UnitName Value="FileMD3"/>
       </Item42>
       <Item43>
-        <Filename Value="../Source/FileOCT.pas"/>
+        <Filename Value="../Source/FileFormats/FileOCT.pas"/>
         <UnitName Value="FileOCT"/>
       </Item43>
       <Item44>
-        <Filename Value="../Source/FileX.pas"/>
+        <Filename Value="../Source/FileFormats/FileX.pas"/>
         <UnitName Value="FileX"/>
       </Item44>
       <Item45>
-        <Filename Value="../Source/GLFile3DS.pas"/>
+        <Filename Value="../Source/FileFormats/GLFile3DS.pas"/>
         <UnitName Value="GLFile3DS"/>
       </Item45>
       <Item46>
-        <Filename Value="../Source/GLFile3DSSceneObjects.pas"/>
+        <Filename Value="../Source/FileFormats/GLFile3DSSceneObjects.pas"/>
         <UnitName Value="GLFile3DSSceneObjects"/>
       </Item46>
       <Item47>
-        <Filename Value="../Source/GLFileASE.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileASE.pas"/>
         <UnitName Value="GLFileASE"/>
       </Item47>
       <Item48>
-        <Filename Value="../Source/GLFileB3D.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileB3D.pas"/>
         <UnitName Value="GLFileB3D"/>
       </Item48>
       <Item49>
-        <Filename Value="../Source/GLFileDDS.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileDDS.pas"/>
         <UnitName Value="GLFileDDS"/>
       </Item49>
       <Item50>
-        <Filename Value="../Source/GLFileGL2.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileGL2.pas"/>
         <UnitName Value="GLFileGL2"/>
       </Item50>
       <Item51>
-        <Filename Value="../Source/GLFileGTS.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileGTS.pas"/>
         <UnitName Value="GLFileGTS"/>
       </Item51>
       <Item52>
-        <Filename Value="../Source/GLFileHDR.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileHDR.pas"/>
         <UnitName Value="GLFileHDR"/>
       </Item52>
       <Item53>
-        <Filename Value="../Source/GLFileJPEG.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileJPEG.pas"/>
         <UnitName Value="GLFileJPEG"/>
       </Item53>
       <Item54>
-        <Filename Value="../Source/GLFileLMTS.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileLMTS.pas"/>
         <UnitName Value="GLFileLMTS"/>
       </Item54>
       <Item55>
-        <Filename Value="../Source/GLFileLWO.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileLWO.pas"/>
         <UnitName Value="GLFileLWO"/>
       </Item55>
       <Item56>
-        <Filename Value="../Source/GLFileMD2.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileMD2.pas"/>
         <UnitName Value="GLFileMD2"/>
       </Item56>
       <Item57>
-        <Filename Value="../Source/GLFileMD3.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileMD3.pas"/>
         <UnitName Value="GLFileMD3"/>
       </Item57>
       <Item58>
-        <Filename Value="../Source/GLFileMD5.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileMD5.pas"/>
         <UnitName Value="GLFileMD5"/>
       </Item58>
       <Item59>
-        <Filename Value="../Source/GLFileMDC.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileMDC.pas"/>
         <UnitName Value="GLFileMDC"/>
       </Item59>
       <Item60>
-        <Filename Value="../Source/GLFileMS3D.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileMS3D.pas"/>
         <UnitName Value="GLFileMS3D"/>
       </Item60>
       <Item61>
-        <Filename Value="../Source/GLFileNMF.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileNMF.pas"/>
         <UnitName Value="GLFileNMF"/>
       </Item61>
       <Item62>
-        <Filename Value="../Source/GLFileNurbs.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileNurbs.pas"/>
         <UnitName Value="GLFileNurbs"/>
       </Item62>
       <Item63>
-        <Filename Value="../Source/GLFileO3TC.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileO3TC.pas"/>
         <UnitName Value="GLFileO3TC"/>
       </Item63>
       <Item64>
-        <Filename Value="../Source/GLFileOBJ.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileOBJ.pas"/>
         <UnitName Value="GLFileOBJ"/>
       </Item64>
       <Item65>
-        <Filename Value="../Source/GLFilePAK.pas"/>
+        <Filename Value="../Source/FileFormats/GLFilePAK.pas"/>
         <UnitName Value="GLFilePAK"/>
       </Item65>
       <Item66>
-        <Filename Value="../Source/GLFilePLY.pas"/>
+        <Filename Value="../Source/FileFormats/GLFilePLY.pas"/>
         <UnitName Value="GLFilePLY"/>
       </Item66>
       <Item67>
-        <Filename Value="../Source/GLFilePNG.pas"/>
+        <Filename Value="../Source/FileFormats/GLFilePNG.pas"/>
         <UnitName Value="GLFilePNG"/>
       </Item67>
       <Item68>
-        <Filename Value="../Source/GLFileSMD.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileSMD.pas"/>
         <UnitName Value="GLFileSMD"/>
       </Item68>
       <Item69>
-        <Filename Value="../Source/GLFileSTL.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileSTL.pas"/>
         <UnitName Value="GLFileSTL"/>
       </Item69>
       <Item70>
-        <Filename Value="../Source/GLFileTIN.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileTIN.pas"/>
         <UnitName Value="GLFileTIN"/>
       </Item70>
       <Item71>
-        <Filename Value="../Source/GLFileVRML.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileVRML.pas"/>
         <UnitName Value="GLFileVRML"/>
       </Item71>
       <Item72>
-        <Filename Value="../Source/GLFileZLIB.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileZLIB.pas"/>
         <UnitName Value="GLFileZLIB"/>
       </Item72>
       <Item73>
@@ -327,31 +327,31 @@
         <UnitName Value="GLStarRecord"/>
       </Item73>
       <Item74>
-        <Filename Value="../Source/GLVfsPAK.pas"/>
+        <Filename Value="../Source/FileFormats/GLVfsPAK.pas"/>
         <UnitName Value="GLVfsPAK"/>
       </Item74>
       <Item75>
-        <Filename Value="../Source/HDRImage.pas"/>
+        <Filename Value="../Source/FileFormats/HDRImage.pas"/>
         <UnitName Value="HDRImage"/>
       </Item75>
       <Item76>
-        <Filename Value="../Source/LWObjects.pas"/>
+        <Filename Value="../Source/FileFormats/LWObjects.pas"/>
         <UnitName Value="LWObjects"/>
       </Item76>
       <Item77>
-        <Filename Value="../Source/LWObjects.txt"/>
+        <Filename Value="../Source/FileFormats/LWObjects.txt"/>
         <Type Value="Text"/>
       </Item77>
       <Item78>
-        <Filename Value="../Source/O3TCImage.pas"/>
+        <Filename Value="../Source/FileFormats/O3TCImage.pas"/>
         <UnitName Value="O3TCImage"/>
       </Item78>
       <Item79>
-        <Filename Value="../Source/Q3BSP.pas"/>
+        <Filename Value="../Source/FileFormats/Q3BSP.pas"/>
         <UnitName Value="Q3BSP"/>
       </Item79>
       <Item80>
-        <Filename Value="../Source/Q3MD3.pas"/>
+        <Filename Value="../Source/FileFormats/Q3MD3.pas"/>
         <UnitName Value="Q3MD3"/>
       </Item80>
       <Item81>
@@ -359,31 +359,31 @@
         <UnitName Value="GLSRGBE"/>
       </Item81>
       <Item82>
-        <Filename Value="../Source/GLFileTGA.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileTGA.pas"/>
         <UnitName Value="GLFileTGA"/>
       </Item82>
       <Item83>
-        <Filename Value="../Source/Types3DS.pas"/>
+        <Filename Value="../Source/FileFormats/Types3DS.pas"/>
         <UnitName Value="Types3DS"/>
       </Item83>
       <Item84>
-        <Filename Value="../Source/Utils3DS.pas"/>
+        <Filename Value="../Source/FileFormats/Utils3DS.pas"/>
         <UnitName Value="Utils3DS"/>
       </Item84>
       <Item85>
-        <Filename Value="../Source/VRMLParser.pas"/>
+        <Filename Value="../Source/FileFormats/VRMLParser.pas"/>
         <UnitName Value="VRMLParser"/>
       </Item85>
       <Item86>
-        <Filename Value="../Source/GLSZLibEx.pas"/>
+        <Filename Value="../Source/FileFormats/GLSZLibEx.pas"/>
         <UnitName Value="GLSZLibEx"/>
       </Item86>
       <Item87>
-        <Filename Value="../Source/GLSZLibExApi.pas"/>
+        <Filename Value="../Source/FileFormats/GLSZLibExApi.pas"/>
         <UnitName Value="GLSZLibExApi"/>
       </Item87>
       <Item88>
-        <Filename Value="../Source/ZLibEx.inc"/>
+        <Filename Value="../Source/FileFormats/ZLibEx.inc"/>
         <Type Value="Include"/>
       </Item88>
       <Item89>
@@ -399,7 +399,7 @@
         <UnitName Value="GLAtmosphere"/>
       </Item91>
       <Item92>
-        <Filename Value="../Source/GLBaseMeshSilhouette.pas"/>
+        <Filename Value="../Source/Basis/GLBaseMeshSilhouette.pas"/>
         <UnitName Value="GLBaseMeshSilhouette"/>
       </Item92>
       <Item93>
@@ -495,11 +495,11 @@
         <UnitName Value="GLFPSMovement"/>
       </Item115>
       <Item116>
-        <Filename Value="../Source/GLGameMenu.pas"/>
+        <Filename Value="../Source/GameAPIs/GLGameMenu.pas"/>
         <UnitName Value="GLGameMenu"/>
       </Item116>
       <Item117>
-        <Filename Value="../Source/GLGeomObjects.pas"/>
+        <Filename Value="../Source/Basis/GLGeomObjects.pas"/>
         <UnitName Value="GLGeomObjects"/>
       </Item117>
       <Item118>
@@ -611,7 +611,7 @@
         <UnitName Value="GLNodes"/>
       </Item144>
       <Item145>
-        <Filename Value="../Source/GLObjects.pas"/>
+        <Filename Value="../Source/Basis/GLObjects.pas"/>
         <UnitName Value="GLObjects"/>
       </Item145>
       <Item146>
@@ -639,7 +639,7 @@
         <UnitName Value="GLPerlinPFX"/>
       </Item151>
       <Item152>
-        <Filename Value="../Source/GLPolyhedron.pas"/>
+        <Filename Value="../Source/Basis/GLPolyhedron.pas"/>
         <UnitName Value="GLPolyhedron"/>
       </Item152>
       <Item153>
@@ -723,23 +723,23 @@
         <UnitName Value="GLSLanguage"/>
       </Item172>
       <Item173>
-        <Filename Value="../Source/GLSLProjectedTextures.pas"/>
+        <Filename Value="../Source/Shaders/GLSLProjectedTextures.pas"/>
         <UnitName Value="GLSLProjectedTextures"/>
       </Item173>
       <Item174>
-        <Filename Value="../Source/GLSLBumpShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLBumpShader.pas"/>
         <UnitName Value="GLSLBumpShader"/>
       </Item174>
       <Item175>
-        <Filename Value="../Source/GLSLDiffuseSpecularShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLDiffuseSpecularShader.pas"/>
         <UnitName Value="GLSLDiffuseSpecularShader"/>
       </Item175>
       <Item176>
-        <Filename Value="../Source/GLSLParameter.pas"/>
+        <Filename Value="../Source/Shaders/GLSLParameter.pas"/>
         <UnitName Value="GLSLParameter"/>
       </Item176>
       <Item177>
-        <Filename Value="../Source/GLSLShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLShader.pas"/>
         <UnitName Value="GLSLShader"/>
       </Item177>
       <Item178>
@@ -747,11 +747,11 @@
         <UnitName Value="GLSmoothNavigator"/>
       </Item178>
       <Item179>
-        <Filename Value="../Source/GLSpatialPartitioning.pas"/>
+        <Filename Value="../Source/Basis/GLSpatialPartitioning.pas"/>
         <UnitName Value="GLSpatialPartitioning"/>
       </Item179>
       <Item180>
-        <Filename Value="../Source/GLState.pas"/>
+        <Filename Value="../Source/Basis/GLState.pas"/>
         <UnitName Value="GLState"/>
       </Item180>
       <Item181>
@@ -759,7 +759,7 @@
         <UnitName Value="GLStrings"/>
       </Item181>
       <Item182>
-        <Filename Value="../Source/GLTeapot.pas"/>
+        <Filename Value="../Source/Basis/GLTeapot.pas"/>
         <UnitName Value="GLTeapot"/>
       </Item182>
       <Item183>
@@ -807,19 +807,19 @@
         <UnitName Value="GLUtils"/>
       </Item193>
       <Item194>
-        <Filename Value="../Source/GLVectorFileObjects.pas"/>
+        <Filename Value="../Source/Basis/GLVectorFileObjects.pas"/>
         <UnitName Value="GLVectorFileObjects"/>
       </Item194>
       <Item195>
-        <Filename Value="../Source/GLVerletClasses.pas"/>
+        <Filename Value="../Source/Basis/GLVerletClasses.pas"/>
         <UnitName Value="GLVerletClasses"/>
       </Item195>
       <Item196>
-        <Filename Value="../Source/GLVerletClothify.pas"/>
+        <Filename Value="../Source/Basis/GLVerletClothify.pas"/>
         <UnitName Value="GLVerletClothify"/>
       </Item196>
       <Item197>
-        <Filename Value="../Source/GLVerletSkeletonColliders.pas"/>
+        <Filename Value="../Source/Basis/GLVerletSkeletonColliders.pas"/>
         <UnitName Value="GLVerletSkeletonColliders"/>
       </Item197>
       <Item198>
@@ -847,7 +847,7 @@
         <UnitName Value="GLFullScreenViewer"/>
       </Item203>
       <Item204>
-        <Filename Value="../Source/GLKeyboard.pas"/>
+        <Filename Value="../Source/Basis/GLKeyboard.pas"/>
         <UnitName Value="GLKeyboard"/>
       </Item204>
       <Item205>
@@ -859,91 +859,91 @@
         <UnitName Value="GLViewer"/>
       </Item206>
       <Item207>
-        <Filename Value="../Source/GLAsmShader.pas"/>
+        <Filename Value="../Source/Shaders/GLAsmShader.pas"/>
         <UnitName Value="GLAsmShader"/>
       </Item207>
       <Item208>
-        <Filename Value="../Source/GLBumpShader.pas"/>
+        <Filename Value="../Source/Shaders/GLBumpShader.pas"/>
         <UnitName Value="GLBumpShader"/>
       </Item208>
       <Item209>
-        <Filename Value="../Source/GLCelShader.pas"/>
+        <Filename Value="../Source/Shaders/GLCelShader.pas"/>
         <UnitName Value="GLCelShader"/>
       </Item209>
       <Item210>
-        <Filename Value="../Source/GLHiddenLineShader.pas"/>
+        <Filename Value="../Source/Shaders/GLHiddenLineShader.pas"/>
         <UnitName Value="GLHiddenLineShader"/>
       </Item210>
       <Item211>
-        <Filename Value="../Source/GLMultiMaterialShader.pas"/>
+        <Filename Value="../Source/Shaders/GLMultiMaterialShader.pas"/>
         <UnitName Value="GLMultiMaterialShader"/>
       </Item211>
       <Item212>
-        <Filename Value="../Source/GLOutlineShader.pas"/>
+        <Filename Value="../Source/Shaders/GLOutlineShader.pas"/>
         <UnitName Value="GLOutlineShader"/>
       </Item212>
       <Item213>
-        <Filename Value="../Source/GLPhongShader.pas"/>
+        <Filename Value="../Source/Shaders/GLPhongShader.pas"/>
         <UnitName Value="GLPhongShader"/>
       </Item213>
       <Item214>
-        <Filename Value="../Source/GLShaderCombiner.pas"/>
+        <Filename Value="../Source/Shaders/GLShaderCombiner.pas"/>
         <UnitName Value="GLShaderCombiner"/>
       </Item214>
       <Item215>
-        <Filename Value="../Source/GLTexCombineShader.pas"/>
+        <Filename Value="../Source/Shaders/GLTexCombineShader.pas"/>
         <UnitName Value="GLTexCombineShader"/>
       </Item215>
       <Item216>
-        <Filename Value="../Source/GLTextureSharingShader.pas"/>
+        <Filename Value="../Source/Shaders/GLTextureSharingShader.pas"/>
         <UnitName Value="GLTextureSharingShader"/>
       </Item216>
       <Item217>
-        <Filename Value="../Source/GLUserShader.pas"/>
+        <Filename Value="../Source/Shaders/GLUserShader.pas"/>
         <UnitName Value="GLUserShader"/>
       </Item217>
       <Item218>
-        <Filename Value="../Source/GLFileBMP.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileBMP.pas"/>
         <UnitName Value="GLFileBMP"/>
       </Item218>
       <Item219>
-        <Filename Value="../Source/TGA.pas"/>
+        <Filename Value="../Source/FileFormats/TGA.pas"/>
         <UnitName Value="TGA"/>
       </Item219>
       <Item220>
-        <Filename Value="../Source/GLSLPostShaders.pas"/>
+        <Filename Value="../Source/Shaders/GLSLPostShaders.pas"/>
         <UnitName Value="GLSLPostShaders"/>
       </Item220>
       <Item221>
-        <Filename Value="../Source/GLSLErosionShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLErosionShader.pas"/>
         <UnitName Value="GLSLErosionShader"/>
       </Item221>
       <Item222>
-        <Filename Value="../Source/GLSLFurShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLFurShader.pas"/>
         <UnitName Value="GLSLFurShader"/>
       </Item222>
       <Item223>
-        <Filename Value="../Source/GLSLGlassShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLGlassShader.pas"/>
         <UnitName Value="GLSLGlassShader"/>
       </Item223>
       <Item224>
-        <Filename Value="../Source/GLSLGoochShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLGoochShader.pas"/>
         <UnitName Value="GLSLGoochShader"/>
       </Item224>
       <Item225>
-        <Filename Value="../Source/GLSLIvoryShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLIvoryShader.pas"/>
         <UnitName Value="GLSLIvoryShader"/>
       </Item225>
       <Item226>
-        <Filename Value="../Source/GLSLSemShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLSemShader.pas"/>
         <UnitName Value="GLSLSemShader"/>
       </Item226>
       <Item227>
-        <Filename Value="../Source/GLSLToonShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLToonShader.pas"/>
         <UnitName Value="GLSLToonShader"/>
       </Item227>
       <Item228>
-        <Filename Value="../Source/GLSLVertexDisplacementShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLVertexDisplacementShader.pas"/>
         <UnitName Value="GLSLVertexDisplacementShader"/>
       </Item228>
       <Item229>
@@ -955,23 +955,23 @@
         <UnitName Value="GLMeshLines"/>
       </Item230>
       <Item231>
-        <Filename Value="../Source/GLSLLatticeShader.pas"/>
+        <Filename Value="../Source/Shaders/GLSLLatticeShader.pas"/>
         <UnitName Value="GLSLLatticeShader"/>
       </Item231>
       <Item232>
-        <Filename Value="../Source/GLFileQ3BSP.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileQ3BSP.pas"/>
         <UnitName Value="GLFileQ3BSP"/>
       </Item232>
       <Item233>
-        <Filename Value="../Source/GLFileOCT.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileOCT.pas"/>
         <UnitName Value="GLFileOCT"/>
       </Item233>
       <Item234>
-        <Filename Value="../Source/GLFileDXF.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileDXF.pas"/>
         <UnitName Value="GLFileDXF"/>
       </Item234>
       <Item235>
-        <Filename Value="../Source/GLFileGRD.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileGRD.pas"/>
         <UnitName Value="GLFileGRD"/>
       </Item235>
       <Item236>
@@ -979,15 +979,24 @@
         <UnitName Value="GLCustomShader"/>
       </Item236>
       <Item237>
-        <Filename Value="../Source/GLScriptBase.pas"/>
+        <Filename Value="../Source/ScriptingAPIs/GLScriptBase.pas"/>
         <UnitName Value="GLScriptBase"/>
       </Item237>
       <Item238>
-        <Filename Value="../Source/GLFileX.pas"/>
+        <Filename Value="../Source/FileFormats/GLFileX.pas"/>
         <AddToUsesPkgSection Value="False"/>
         <UnitName Value="GLFileX"/>
       </Item238>
+      <Item239>
+        <Filename Value="../Source/Plateform/GLLCLViewer.pas"/>
+        <UnitName Value="GLLCLViewer"/>
+      </Item239>
+      <Item240>
+        <Filename Value="../Source/DesignTime/GLSceneForm.pas"/>
+        <UnitName Value="GLSceneForm"/>
+      </Item240>
     </Files>
+    <CompatibilityMode Value="True"/>
     <RequiredPkgs Count="3">
       <Item1>
         <PackageName Value="ImagesForLazarus"/>
@@ -1007,7 +1016,6 @@
     </UsageOptions>
     <PublishOptions>
       <Version Value="2"/>
-      <IgnoreBinaries Value="False"/>
     </PublishOptions>
   </Package>
 </CONFIG>
Index: Packages/GLScene_RunTime.pas
===================================================================
--- Packages/GLScene_RunTime.pas	(revision 7332)
+++ Packages/GLScene_RunTime.pas	(working copy)
@@ -1,61 +1,61 @@
-{ This file was automatically created by Lazarus. Do not edit!
-  This source is only used to compile and install the package.
- }
-
-unit GLScene_RunTime;
-
-{$warn 5023 off : no warning about unused units}
-interface
-
-uses
-  GLApplicationFileIO, GLAsyncTimer, GLBaseClasses, GLBumpMapping, 
-  GLCurvesAndSurfaces, GLGeometryBB, GLGeometryCoordinates, GLCanvas, GLColor, 
-  GLCoordinates, GLPipelineTransformation, GLSCrossXML, GLSLog, 
-  GLTextureCombiners, GLHeightTileFile, GLImageUtils, GLMeshUtils, GLOctree, 
-  OpenGLAdapter, OpenGLTokens, GLPerlinNoise3D, GLPersistentClasses, 
-  GLPolynomials, GLSpacePartition, GLSpline, GLVectorGeometry, GLVectorLists, 
-  GLVectorTypes, GLVerletTypes, GLVerletHairClasses, GLXCollection, XOpenGL, 
-  Const3DS, DDSImage, DXTC, File3DS, FileB3D, FileGL2, FileMD2, FileMD3, 
-  FileOCT, FileX, GLFile3DS, GLFile3DSSceneObjects, GLFileASE, GLFileB3D, 
-  GLFileDDS, GLFileGL2, GLFileGTS, GLFileHDR, GLFileJPEG, GLFileLMTS, 
-  GLFileLWO, GLFileMD2, GLFileMD3, GLFileMD5, GLFileMDC, GLFileMS3D, 
-  GLFileNMF, GLFileNurbs, GLFileO3TC, GLFileOBJ, GLFilePAK, GLFilePLY, 
-  GLFilePNG, GLFileSMD, GLFileSTL, GLFileTIN, GLFileVRML, GLFileZLIB, 
-  GLStarRecord, GLVfsPAK, HDRImage, LWObjects, O3TCImage, Q3BSP, Q3MD3, 
-  GLSRGBE, GLFileTGA, Types3DS, Utils3DS, VRMLParser, GLSZLibEx, GLSZLibExApi, 
-  GLAnimatedSprite, GLAsyncHDS, GLAtmosphere, GLBaseMeshSilhouette, 
-  GLBehaviours, GLBitmapFont, GLBlur, GLBSP, GLBumpmapHDS, GLCadencer, 
-  GLCameraController, GLCollision, GLCompositeImage, GLConsole, GLContext, 
-  GLDCE, GLDCEMisc, GLDynamicTexture, GLEllipseCollision, 
-  GLEParticleMasksManager, GLExplosionFx, GLExtrusion, GLFBO, GLFBORenderer, 
-  GLFeedback, GLFireFX, GLFPSMovement, GLGameMenu, GLGeomObjects, GLGizmo, 
-  GLGizmoEx, GLGLUTesselation, GLGraph, GLGraphics, GLGui, GLHeightData, 
-  GLHeightTileFileHDS, GLHUDObjects, GLImposter, GLLensFlare, GLLinePFX, 
-  GLManager, GLMaterial, GLMaterialMultiProxy, GLMaterialScript, GLMesh, 
-  GLMeshBuilder, GLMeshCSG, GLMeshOptimizer, GLMirror, GLMovement, 
-  GLMultiPolygon, GLMultiProxy, GLMultisampleImage, GLNavigator, GLNodes, 
-  GLObjects, GLParametricSurfaces, GLParticleFX, GLParticles, GLPerlin, 
-  GLPerlinBase, GLPerlinPFX, GLPolyhedron, GLPortal, GLPostEffects, 
-  GLProcTextures, GLProjectedTextures, GLProxyObjects, GLRagdoll, 
-  GLRenderContextInfo, GLROAMPatch, GLSArchiveManager, GLScene, GLSelection, 
-  GLShadowHDS, GLShadowPlane, GLShadowVolume, GLSilhouette, 
-  GLSimpleNavigation, GLSkyBox, GLSkydome, GLSLanguage, GLSLProjectedTextures, 
-  GLSLBumpShader, GLSLDiffuseSpecularShader, GLSLParameter, GLSLShader, 
-  GLSmoothNavigator, GLSpatialPartitioning, GLState, GLStrings, GLTeapot, 
-  GLTerrainRenderer, GLTexLensFlare, GLTexture, GLTexturedHDS, 
-  GLTextureFormat, GLThorFX, GLTilePlane, GLTimeEventsMgr, GLTrail, GLTree, 
-  GLUtils, GLVectorFileObjects, GLVerletClasses, GLVerletClothify, 
-  GLVerletSkeletonColliders, GLWaterPlane, GLWindows, GLWindowsFont, 
-  GLzBuffer, GLCrossPlatform, GLFullScreenViewer, GLKeyboard, GLScreen, 
-  GLViewer, GLAsmShader, GLBumpShader, GLCelShader, GLHiddenLineShader, 
-  GLMultiMaterialShader, GLOutlineShader, GLPhongShader, GLShaderCombiner, 
-  GLTexCombineShader, GLTextureSharingShader, GLUserShader, GLFileBMP, TGA, 
-  GLSLPostShaders, GLSLErosionShader, GLSLFurShader, GLSLGlassShader, 
-  GLSLGoochShader, GLSLIvoryShader, GLSLSemShader, GLSLToonShader, 
-  GLSLVertexDisplacementShader, GLTriangulation, GLMeshLines, 
-  GLSLLatticeShader, GLFileQ3BSP, GLFileOCT, GLFileDXF, GLFileGRD, 
-  GLCustomShader, GLScriptBase;
-
-implementation
-
-end.
+{ This file was automatically created by Lazarus. Do not edit!
+  This source is only used to compile and install the package.
+ }
+
+unit GLScene_RunTime;
+
+{$warn 5023 off : no warning about unused units}
+interface
+
+uses
+  GLApplicationFileIO, GLAsyncTimer, GLBaseClasses, GLBumpMapping, 
+  GLCurvesAndSurfaces, GLGeometryBB, GLGeometryCoordinates, GLCanvas, GLColor, 
+  GLCoordinates, GLPipelineTransformation, GLSCrossXML, GLSLog, 
+  GLTextureCombiners, GLHeightTileFile, GLImageUtils, GLMeshUtils, GLOctree, 
+  OpenGLAdapter, OpenGLTokens, GLPerlinNoise3D, GLPersistentClasses, 
+  GLPolynomials, GLSpacePartition, GLSpline, GLVectorGeometry, GLVectorLists, 
+  GLVectorTypes, GLVerletTypes, GLVerletHairClasses, GLXCollection, XOpenGL, 
+  Const3DS, DDSImage, DXTC, File3DS, FileB3D, FileGL2, FileMD2, FileMD3, 
+  FileOCT, FileX, GLFile3DS, GLFile3DSSceneObjects, GLFileASE, GLFileB3D, 
+  GLFileDDS, GLFileGL2, GLFileGTS, GLFileHDR, GLFileJPEG, GLFileLMTS, 
+  GLFileLWO, GLFileMD2, GLFileMD3, GLFileMD5, GLFileMDC, GLFileMS3D, 
+  GLFileNMF, GLFileNurbs, GLFileO3TC, GLFileOBJ, GLFilePAK, GLFilePLY, 
+  GLFilePNG, GLFileSMD, GLFileSTL, GLFileTIN, GLFileVRML, GLFileZLIB, 
+  GLStarRecord, GLVfsPAK, HDRImage, LWObjects, O3TCImage, Q3BSP, Q3MD3, 
+  GLSRGBE, GLFileTGA, Types3DS, Utils3DS, VRMLParser, GLSZLibEx, GLSZLibExApi, 
+  GLAnimatedSprite, GLAsyncHDS, GLAtmosphere, GLBaseMeshSilhouette, 
+  GLBehaviours, GLBitmapFont, GLBlur, GLBSP, GLBumpmapHDS, GLCadencer, 
+  GLCameraController, GLCollision, GLCompositeImage, GLConsole, GLContext, 
+  GLDCE, GLDCEMisc, GLDynamicTexture, GLEllipseCollision, 
+  GLEParticleMasksManager, GLExplosionFx, GLExtrusion, GLFBO, GLFBORenderer, 
+  GLFeedback, GLFireFX, GLFPSMovement, GLGameMenu, GLGeomObjects, GLGizmo, 
+  GLGizmoEx, GLGLUTesselation, GLGraph, GLGraphics, GLGui, GLHeightData, 
+  GLHeightTileFileHDS, GLHUDObjects, GLImposter, GLLensFlare, GLLinePFX, 
+  GLManager, GLMaterial, GLMaterialMultiProxy, GLMaterialScript, GLMesh, 
+  GLMeshBuilder, GLMeshCSG, GLMeshOptimizer, GLMirror, GLMovement, 
+  GLMultiPolygon, GLMultiProxy, GLMultisampleImage, GLNavigator, GLNodes, 
+  GLObjects, GLParametricSurfaces, GLParticleFX, GLParticles, GLPerlin, 
+  GLPerlinBase, GLPerlinPFX, GLPolyhedron, GLPortal, GLPostEffects, 
+  GLProcTextures, GLProjectedTextures, GLProxyObjects, GLRagdoll, 
+  GLRenderContextInfo, GLROAMPatch, GLSArchiveManager, GLScene, GLSelection, 
+  GLShadowHDS, GLShadowPlane, GLShadowVolume, GLSilhouette, 
+  GLSimpleNavigation, GLSkyBox, GLSkydome, GLSLanguage, GLSLProjectedTextures, 
+  GLSLBumpShader, GLSLDiffuseSpecularShader, GLSLParameter, GLSLShader, 
+  GLSmoothNavigator, GLSpatialPartitioning, GLState, GLStrings, GLTeapot, 
+  GLTerrainRenderer, GLTexLensFlare, GLTexture, GLTexturedHDS, 
+  GLTextureFormat, GLThorFX, GLTilePlane, GLTimeEventsMgr, GLTrail, GLTree, 
+  GLUtils, GLVectorFileObjects, GLVerletClasses, GLVerletClothify, 
+  GLVerletSkeletonColliders, GLWaterPlane, GLWindows, GLWindowsFont, 
+  GLzBuffer, GLCrossPlatform, GLFullScreenViewer, GLKeyboard, GLScreen, 
+  GLViewer, GLAsmShader, GLBumpShader, GLCelShader, GLHiddenLineShader, 
+  GLMultiMaterialShader, GLOutlineShader, GLPhongShader, GLShaderCombiner, 
+  GLTexCombineShader, GLTextureSharingShader, GLUserShader, GLFileBMP, TGA, 
+  GLSLPostShaders, GLSLErosionShader, GLSLFurShader, GLSLGlassShader, 
+  GLSLGoochShader, GLSLIvoryShader, GLSLSemShader, GLSLToonShader, 
+  GLSLVertexDisplacementShader, GLTriangulation, GLMeshLines, 
+  GLSLLatticeShader, GLFileQ3BSP, GLFileOCT, GLFileDXF, GLFileGRD, 
+  GLCustomShader, GLScriptBase, GLLCLViewer, GLSceneForm;
+
+implementation
+
+end.
Index: Source/DesignTime/GLSceneForm.pas
===================================================================
--- Source/DesignTime/GLSceneForm.pas	(revision 7332)
+++ Source/DesignTime/GLSceneForm.pas	(working copy)
@@ -139,6 +139,7 @@
   public
     constructor Create(AOwner: TComponent); override;
     destructor Destroy; override;
+    class procedure WSRegisterClass; override;
     procedure DestroyWnd; override;
 
     property IsRenderingContextAvailable: Boolean read
@@ -219,6 +220,17 @@
   inherited Destroy;
 end;
 
+class procedure TGLSceneForm.WSRegisterClass;
+const
+  Registered : Boolean = False;
+begin
+  if Registered then
+    Exit;
+  inherited WSRegisterClass;
+  RegisterWSComponent(TGLSceneForm, TGLSOpenGLForm);
+  Registered := True;
+end;
+
 // Notification
 //
 
@@ -793,7 +805,7 @@
 
 procedure GLRegisterWSComponent(aControl: TComponentClass);
 begin
-  RegisterWSComponent(aControl, TGLSOpenGLForm);
+//  RegisterWSComponent(aControl, TGLSOpenGLForm);
 end;
 {$endif}
 
Index: Source/GLViewer.pas
===================================================================
--- Source/GLViewer.pas	(revision 7332)
+++ Source/GLViewer.pas	(working copy)
@@ -29,14 +29,19 @@
 {$I GLScene.inc}
 
 uses
-  GLContext,
+  GLContext, WSLCLClasses, GLWidgetContext,
 
   GLLCLViewer;
 type
 
-  TGLSceneViewer = GLLCLViewer.TGLSceneViewer;
+  { TGLSceneViewer }
 
+  TGLSceneViewer = class(GLLCLViewer.TGLSceneViewer)
+  public
+    class procedure WSRegisterClass; override;
+  end;
 
+
 procedure SetupVSync(const AVSyncMode : TVSyncMode);
 
 implementation
@@ -60,6 +65,20 @@
     end;
   end;
 end;
+
+{ TGLSceneViewer }
+
+class procedure TGLSceneViewer.WSRegisterClass;
+const
+  Registered : Boolean = False;
+begin
+  if Registered then
+    Exit;
+  inherited WSRegisterClass;
+  RegisterWSComponent(TGLSceneViewer, TGLSOpenGLControl);
+  Registered := True;
+end;
+
 {$ENDIF}
 {$IFDEF Linux}
 begin
Index: Source/Plateform/GLWidgetContext.pas
===================================================================
--- Source/Plateform/GLWidgetContext.pas	(revision 7332)
+++ Source/Plateform/GLWidgetContext.pas	(working copy)
@@ -258,7 +258,7 @@
 
 procedure GLRegisterWSComponent(aControl: TComponentClass);
 begin
-  RegisterWSComponent(aControl, TGLSOpenGLControl);
+//  RegisterWSComponent(aControl, TGLSOpenGLControl);
 end;
 {$ENDIF}
 
