Index: Source/GLCarbonContext.pas
===================================================================
--- Source/GLCarbonContext.pas	(révision 43)
+++ Source/GLCarbonContext.pas	(copie de travail)
@@ -87,8 +87,7 @@
 // ------------------------------------------------------------------
 // -----------------------------------------------------------------
 uses
-  GLState
-  {$IFDEF GLS_LOGGING}, GLSLog {$ENDIF};
+  GLState, GLSLog;
 
 
 resourcestring
