Index: Source/GLSLog.pas
===================================================================
--- Source/GLSLog.pas	(révision 6416)
+++ Source/GLSLog.pas	(copie de travail)
@@ -608,8 +608,7 @@
       begin
         {$IFDEF FPC}
         if MessageDlg(Format(lErrorMessage, [lLogOriginalDir + sRec.Name,
-          GetLastOSErrorGetLastError]),
-          mtWarning, mbYesNo, -1, mbYes = mrNo
+          GetLastOSError]), mtWarning, mbYesNo, -1) = mrNo 
           then Break;
         {$ELSE}
         if MessageDlg(Format(lErrorMessage, [lLogOriginalDir + sRec.Name,
Index: Source/GLWindowsFont.pas
===================================================================
--- Source/GLWindowsFont.pas	(révision 6416)
+++ Source/GLWindowsFont.pas	(copie de travail)
@@ -312,6 +312,9 @@
   bitmap: TGLBitmap;
   ch: widechar;
   i, cw, nbChars, n: Integer;
+  {$IFDEF FPC}
+  ms: TMemoryStream;
+  {$ENDIF}
 begin
   InvalidateUsers;
   Glyphs.OnChange := nil;
@@ -362,6 +365,13 @@
   bitmap.Width := TextureWidth;
 
   ComputeCharRects(bitmap);
+  {$IFDEF FPC}
+    ms:=TMemoryStream.Create;
+    bitmap.SaveToStream(ms);
+    ms.Position:=0;
+    bitmap.LoadFromStream(ms);
+    ms.free;
+  {$ENDIF}
   FCharsLoaded := true;
   Glyphs.OnChange := OnGlyphsChanged;
 end;
